broker/lib/controllers/ServiceBrokerApiController.js

```diff
@@ -105,33 +150,108 @@ class ServiceBrokerApiController extends FabrikBaseController {
       .then(done);
   }
 
+  patchInstanceV2(req, res) {
+    const params = _
+      .chain(req.body)
+      .omit('plan_id', 'service_id')
+      .cloneDeep()
+      .value();
+    //cloning here so that the DirectorInstance.update does not unset the 'service-fabrik-operation' from original req.body object
+
+    function done(result) {
+      let statusCode = CONST.HTTP_STATUS_CODE.OK;
+      const body = {};
+      if (req.instance.async) {
+        statusCode = CONST.HTTP_STATUS_CODE.ACCEPTED;
+        body.operation = utils.encodeBase64(result);
+      } else if (result && result.description) {
+        body.description = result.description;
+      }
+      res.status(statusCode).send(body);
+    }
+
+    req.operation_type = CONST.OPERATION_TYPE.UPDATE;
+    return Promise
+      .try(() => {
+        if (!req.manager.isUpdatePossible(params.previous_values.plan_id)) {
+          throw new BadRequest(`Update to plan '${req.manager.plan.name}' is not possible`);
+        }
+        return req.instance.update(params);
+      })
+      .then(done)
+      .catch(err => {
+        return lockManager.unlock(req.params.instance_id)
+          .throw(err);
+      });
+  }
+
   deleteInstance(req, res) {
+    if (config.enable_service_fabrik_v2) {
+      return this.deleteInstanceV2(req, res);
+    }
+    return this.deleteInstanceV1(req, res);
+  }
+
+  deleteInstanceV1(req, res) {
     const params = _.omit(req.query, 'plan_id', 'service_id');
 
     function done(result) {
-      let statusCode = 200;
+      let statusCode = CONST.HTTP_STATUS_CODE.OK;
       const body = {};
       if (req.instance.async) {
-        statusCode = 202;
+        statusCode = CONST.HTTP_STATUS_CODE.ACCEPTED;
         body.operation = utils.encodeBase64(result);
       }
       res.status(statusCode).send(body);
     }
 
     function gone(err) {
       /* jshint unused:false */
-      res.status(410).send({});
+      res.status(CONST.HTTP_STATUS_CODE.GONE).send({});
     }
     req.operation_type = CONST.OPERATION_TYPE.DELETE;
-    this.validateRequest(req, res);
 
     return Promise
       .try(() => req.instance.delete(params))
       .then(done)
       .catch(ServiceInstanceNotFound, gone);
   }
 
+  deleteInstanceV2(req, res) {
+    const params = _.omit(req.query, 'plan_id', 'service_id');
+
+    function done(result) {
+      let statusCode = CONST.HTTP_STATUS_CODE.OK;
+      const body = {};
+      if (req.instance.async) {
+        statusCode = CONST.HTTP_STATUS_CODE.ACCEPTED;
+        body.operation = utils.encodeBase64(result);
+      }
+      res.status(statusCode).send(body);
+    }
+
+    function gone(err) {
+      /* jshint unused:false */
+      res.status(CONST.HTTP_STATUS_CODE.GONE).send({});
+    }
+    req.operation_type = CONST.OPERATION_TYPE.DELETE;
+    return req.instance.delete(params)
+      .then(done)
+      .catch(err => {
```

Comment:
same comment as above with regards to unlocking. So if this unlocking code is removed then essentially code for deleteInstanceV1 and V2 remains same. Same would be the case for even create/udpate methods as well.